# Data Analysis
#### John R. Starr; jrs294@pitt.edu
Time for some analysis! What's on the schedule for today?

### Table of Contents
- [Section 1: Loading in Files](#Section-1:-Loading-in-Files)
- [Section 2: Examining SOV Data](#Section-2:-Examining-SOV-Data)
- [Section 3: Examining SVO Data](#Section-3:-Examining-SOV-Data)
- [Section 4: Examining Other Data](#Section-4:-Examining-Other-Data)
- [Section 5: Conclusions](#Section-5:-Conclusions)

### Section 1: Loading in Files

Like usual, we're going to load in all the usual modules and DFs:

In [1]:
import nltk
import numpy as np
import pandas as pd

In [2]:
# Releasing all output:
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [3]:
df = pd.read_pickle('only_ordered_final.pkl')

A breakdown on what we've got:

In [4]:
df.head()

ID                                                Eng  \
4    5                                   stop please stop   
5    6    you have a week  evans then well burn the house   
8    9                          god damn it put that down   
10  11                         its the last feed weve got   
13  14  herds over the ridge by now you go get cleaned up   

                                                  Far  \
4             دست نگه داريد خواهش ميکنم دست نگه داريد   
5   اوانز تو فقط يک هفته وقت داري وگرنه خونتو خواه...   
8                           لعنت به تو اونو بذار زمين   
10                اين آخرين علوفه اي بود که ما داشتيم   
13    گله را آوردم بيرون الان تو برو اونجا را تميز کن   

                                              Eng_Tok  \
4                                [stop, please, stop]   
5   [you, have, a, week, evans, then, well, burn, ...   
8                    [god, damn, it, put, that, down]   
10                  [its, the, last, feed, weve, got]   
13  [herds, over, the, ridge, by, now, you, go, ge...   

                                              Far_Tok  Eng_Len  Far_Len  \
4    [دست, نگه, داريد, خواهش, ميکنم, دست, نگه, داريد]        3        8   
5   [اوانز, تو, فقط, يک, هفته, وقت, داري, وگرنه, خ...       10       11   
8                    [لعنت, به, تو, اونو, بذار, زمين]        6        6   
10       [اين, آخرين, علوفه, اي, بود, که, ما, داشتيم]        6        8   
13  [گله, را, آوردم, بيرون, الان, تو, برو, اونجا, ...       11       11   

                                            Eng_Types  \
4                                      {stop, please}   
5   {then, the, well, burn, house, evans, week, yo...   
8                    {damn, put, that, it, god, down}   
10                  {feed, its, the, weve, got, last}   
13  {the, by, over, up, get, now, go, ridge, you, ...   

                                            Far_Types  \
4                     {داريد, خواهش, ميکنم, نگه, دست}   
5   {وقت, وگرنه, خواهيم, سوزوند, داري, خونتو, فقط,...   
8                    {بذار, اونو, لعنت, تو, زمين, به}   
10       {داشتيم, بود, اي, آخرين, ما, اين, که, علوفه}   
13  {گله, بيرون, اونجا, تو, آوردم, برو, الان, تميز...   

                                              Far_POS  \
4   [(دست, N), (نگه, N), (داريد, V), (خواهش, Ne), ...   
5   [(اوانز, Ne), (تو, PRO), (فقط, ADV), (يک, Ne),...   
8   [(لعنت, N), (به, P), (تو, PRO), (اونو, PRO), (...   
10  [(اين, Ne), (آخرين, NUM), (علوفه, N), (اي, N),...   
13  [(گله, N), (را, POSTP), (آوردم, V), (بيرون, CO...   

                                           Far_Chunks  \
4   [دست NP] [نگه داريد VP] [خواهش ميکنم دست NP] [...   
5   [اوانز تو NP] [فقط يک هفته وقت ADVP] [داري VP]...   
8   [لعنت NP] [به PP] [تو NP] [اونو NP] [بذار VP] ...   
10  [اين آخرين علوفه NP] [اي NP] [بود VP] که [ما N...   
13  [گله NP] [را POSTP] [آوردم VP] بيرون [الان ADV...   

                                              Eng_POS  \
4              [(stop, JJ), (please, NN), (stop, VB)]   
5   [(you, PRP), (have, VBP), (a, DT), (week, NN),...   
8   [(god, NN), (damn, VBZ), (it, PRP), (put, VBD)...   
10  [(its, PRP$), (the, DT), (last, JJ), (feed, NN...   
13  [(herds, NNS), (over, IN), (the, DT), (ridge, ...   

                                           Eng_Chunks Word_Order_Final  
4    [[(stop, JJ), (please, NN)], [[('stop', 'VB')]]]              SVO  
5   [(you, PRP), [[('have', 'VBP')], [('a', 'DT'),...              SVO  
8   [[(god, NN)], [[('damn', 'VBZ')]], (it, PRP), ...              SOV  
10  [(its, PRP$), [(the, DT), (last, JJ), (feed, N...              SOV  
13  [(herds, NNS), [[('over', 'IN')], [('the', 'DT...              SVO

Next, we're going to reset the index inplace, to make it a little easier to navigate the DF!

In [5]:
df.reset_index(drop = True, inplace = True)

What are some basic statistics about our data?

In [6]:
print(len(df))
print()
print(df['Word_Order_Final'].value_counts())

189561

SOV    120607
SVO     68954
Name: Word_Order_Final, dtype: int64


Well, we have nearly double the number of SOV sentences than we do SVO sentences -- this isn't a bad thing, since Persian is underlyingly SOV. Let's separate the different orderings into their own respective DFs and then examine some of the structures that we find in both.

In [7]:
#  Sorting by the values
df.sort_values(by=['Word_Order_Final'], inplace = True)

In [8]:
# Creating new DFs
sov_df = df.loc[df.Word_Order_Final == 'SOV']
svo_df = df.loc[df.Word_Order_Final == 'SVO']

In [9]:
# Making sure they are the proper lengths
print(len(sov_df))
print(len(svo_df))

120607
68954


Cool! Let's start with the SOV data first.

### Section 2: Examining SOV Data
Just a snippet of what we're working with:

In [10]:
sov_df.head()

ID                                      Eng  \
94780   276464  i will never sell you my land your land   
157247  459494       i'm going to cut him down.  don't.   
89398   261678     oh  my gosh i know i cant beiieve it   
157249  459500                          danielle don't.   
89396   261676             hey  you guys what this mean   

                                          Far  \
94780    من هيچوقت زمينم را نميفروشم زمينت را   
157247              ميارمش پايين اين كارو نكن   
89398   اوه خداي من ميدونم ، نميتونم باور کنم   
157249                   دانيل ، اين كارو نكن   
89396                 هي ، شماها معني اين چيه   

                                                  Eng_Tok  \
94780   [i, will, never, sell, you, my, land, your, land]   
157247  [i, 'm, going, to, cut, him, down, ., do, n't, .]   
89398       [oh, my, gosh, i, know, i, cant, beiieve, it]   
157249                             [danielle, do, n't, .]   
89396                  [hey, you, guys, what, this, mean]   

                                               Far_Tok  Eng_Len  Far_Len  \
94780     [من, هيچوقت, زمينم, را, نميفروشم, زمينت, را]        9        7   
157247                 [ميارمش, پايين, اين, كارو, نكن]       11        5   
89398   [اوه, خداي, من, ميدونم, ،, نميتونم, باور, کنم]        9        8   
157249                      [دانيل, ،, اين, كارو, نكن]        4        5   
89396                   [هي, ،, شماها, معني, اين, چيه]        6        6   

                                             Eng_Types  \
94780      {will, land, your, my, sell, never, you, i}   
157247  {to, going, cut, 'm, him, n't, do, i, ., down}   
89398       {my, it, gosh, beiieve, know, i, cant, oh}   
157249                          {do, ., danielle, n't}   
89396               {this, what, guys, mean, you, hey}   

                                             Far_Types  \
94780         {هيچوقت, زمينت, نميفروشم, من, زمينم, را}   
157247                 {نكن, پايين, اين, ميارمش, كارو}   
89398   {باور, خداي, ،, نميتونم, من, کنم, ميدونم, اوه}   
157249                      {نكن, ،, دانيل, اين, كارو}   
89396                   {،, معني, شماها, چيه, هي, اين}   

                                                  Far_POS  \
94780   [(من, PRO), (هيچوقت, CONJ), (زمينم, N), (را, P...   
157247  [(ميارمش, N), (پايين, Pe), (اين, Ne), (كارو, N...   
89398   [(اوه, RES), (خداي, Pe), (من, PRO), (ميدونم, V...   
157249  [(دانيل, N), (،, PUNC), (اين, Ne), (كارو, N), ...   
89396   [(هي, N), (،, PUNC), (شماها, PRO), (معني, N), ...   

                                               Far_Chunks  \
94780   [من NP] هيچوقت [زمينم NP] [را POSTP] [نميفروشم...   
157247      [ميارمش NP] [پايين PP] [اين كارو NP] [نكن VP]   
89398   [اوه NP] [خداي من NP] [ميدونم VP] ، [نميتونم V...   
157249                [دانيل NP] ، [اين كارو NP] [نكن VP]   
89396              [هي NP] ، [شماها NP] [معني اين VP] چيه   

                                                  Eng_POS  \
94780   [(i, NN), (will, MD), (never, RB), (sell, VB),...   
157247  [(i, NN), ('m, VBP), (going, VBG), (to, TO), (...   
89398   [(oh, UH), (my, PRP$), (gosh, NN), (i, NN), (k...   
157249    [(danielle, NNS), (do, VBP), (n't, RB), (., .)]   
89396   [(hey, NN), (you, PRP), (guys, VBP), (what, WP...   

                                               Eng_Chunks Word_Order_Final  
94780   [[(i, NN)], (will, MD), (never, RB), [[('sell'...              SOV  
157247  [[(i, NN)], [[("'m", 'VBP')]], [[('going', 'VB...              SOV  
89398   [(oh, UH), (my, PRP$), [(gosh, NN), (i, NN)], ...              SOV  
157249  [(danielle, NNS), [[('do', 'VBP')]], (n't, RB)...              SOV  
89396   [[(hey, NN)], (you, PRP), [[('guys', 'VBP')]],...              SOV

In [11]:
# Making the indexes easier to work with
sov_df.reset_index(drop = True, inplace = True)

Just out of curiousity, I want to see if there's any big difference between the average lengths of the English and Persian SOV sentences:

In [12]:
avg_eng_len = []
for ln in sov_df['Eng_Len']:
    avg_eng_len.append(ln)
print('The average English sentence length in the SOV file is...', (sum(avg_eng_len))/len(sov_df))

The average English sentence length in the SOV file is... 8.797441276211165


In [13]:
avg_far_len = []
for ln in sov_df['Far_Len']:
    avg_far_len.append(ln)
print('The average Persian sentence length in the SOV file is...', (sum(avg_far_len))/len(sov_df))

The average Persian sentence length in the SOV file is... 8.894118915154179


Not too big of a difference, it seems. I figured this wouldn't have much of an impact, but it was worth at least checking.

Let's see how sentences involving the word phrase "is like" fared! This phrase is common (and easily searchable) way of looking for similes.

In [14]:
is_like_bool = []
for line in sov_df['Eng']:
    if 'is like' in line:
        is_like_bool.append(True)
    else:
        is_like_bool.append(False)
has_like = pd.Series(is_like_bool)

In [15]:
like_sov_df = sov_df[has_like]

In [16]:
print(len(like_sov_df))
like_sov_df.head(3)

56


ID                                                Eng  \
14003  272686         capricorns castle is like a fivestar hotel   
17108  206774  i know what it is like to try and scrape by wi...   
17691  204305  having an ulcer is like having a burglar alarm...   

                                                     Far  \
14003              قلعه كاپريكورن مثل يک هتل 5 ستاره است   
17108  من ميدونم كه زندگي بدون پشتيبان چه جون كندني ميشه   
17691  يه چيز حسابي حول و حوش زخم معده م انگار يک دزد...   

                                                 Eng_Tok  \
14003  [capricorns, castle, is, like, a, fivestar, ho...   
17108  [i, know, what, it, is, like, to, try, and, sc...   
17691  [having, an, ulcer, is, like, having, a, burgl...   

                                                 Far_Tok  Eng_Len  Far_Len  \
14003     [قلعه, كاپريكورن, مثل, يک, هتل, 5, ستاره, است]        7        8   
17108  [من, ميدونم, كه, زندگي, بدون, پشتيبان, چه, جون...       14       10   
17691  [يه, چيز, حسابي, حول, و, حوش, زخم, معده, م, ان...       13       19   

                                               Eng_Types  \
14003  {fivestar, hotel, is, castle, like, a, caprico...   
17108  {and, by, to, without, scrape, it, is, what, l...   
17691  {burglar, having, is, like, off, inside, ulcer...   

                                               Far_Types  \
14003     {5, است, يک, ستاره, كاپريكورن, مثل, هتل, قلعه}   
17108  {زندگي, ميشه, چه, بدون, من, ميدونم, كه, پشتيبا...   
17691  {بياد, و, زخم, وجود, انگار, صدا, يک, حول, در, ...   

                                                 Far_POS  \
14003  [(قلعه, Ne), (كاپريكورن, N), (مثل, ADVe), (يک,...   
17108  [(من, PRO), (ميدونم, Ne), (كه, DET), (زندگي, N...   
17691  [(يه, N), (چيز, ADV), (حسابي, N), (حول, N), (و...   

                                              Far_Chunks  \
14003  [قلعه كاپريكورن NP] [مثل PP] [يک هتل NP] [5 ست...   
17108  [من NP] [ميدونم كه زندگي NP] [بدون PP] [پشتيبا...   
17691  [يه NP] [چيز ADVP] [حسابي NP] [حول PP] [و حوش ...   

                                                 Eng_POS  \
14003  [(capricorns, NNS), (castle, NN), (is, VBZ), (...   
17108  [(i, NN), (know, VBP), (what, WP), (it, PRP), ...   
17691  [(having, VBG), (an, DT), (ulcer, NN), (is, VB...   

                                              Eng_Chunks Word_Order_Final  
14003  [(capricorns, NNS), [(castle, NN)], [[('is', '...              SOV  
17108  [[(i, NN)], [[('know', 'VBP')]], (what, WP), (...              SOV  
17691  [[[('having', 'VBG')], [('an', 'DT'), ('ulcer'...              SOV

This is a really small DF, with only 56 entires. Perhaps this is because "is like" is more common as a written phrase, rather than a verbal one.

Regardless of the size of the DF, it is clear to see some patterns. Persian's functional equivalents to the words "like" and "that" are "مثل" and "که" respectively. If you look closely at the data, nearly all the Persian sentences in this dataset use those words! And, conveniently enough, you see that those same sentences maintain the SOV structure that we anticipated before.

Let's look at some of the sentences in this set of sentences:

In [17]:
for sample in like_sov_df['Far_Chunks'].sample(5):
    print(sample)
    print()

[اين NP] [مثل PP] [سرپرست هست، NP] [کسي VP] که [سخت ADVP] [کار مي NP] [کنه VP] تا [شمشير NP] [درست ADJP] [کنه VP] .

[اینکه NP] [دنیا NP] [مثل PP] [یک کاتالوگ NP] [باشه VP]

[تسليم شدن NP] [به PP] [بو NP] [يه VP] [او NP] [مانند PP] [تسليم شدن يه NP] [هان NP] [است VP]

[مطالعه طبیعت NP] [این طور NP] [است VP]

[قلعه كاپريكورن NP] [مثل PP] [يک هتل NP] [5 ستاره NP] [است VP]



Most of these sentences use مثل and که to produce similes, but still maintain the SOV structure; the first, second, and fifth sentences, which utilize these functional words, mean the following: 
- 1 "He is like a guardian, someone that works to protect completely with a sword."
- 2 "This world is like a catalog"

However, the third and fourth sentences do not use either مثل or که to produce idiomatic expressions, preferring instead the path of direct metaphor over similes. It will be interesting to see if this is a trend that carries over to the other word orders.

In the meantime, let's search the Persian data for these two functional words and see if there's anything interesting that can be drawn. Unfortunately, I could not figure out a way to specify what a "simile" context would be, so I just raw-searched for مثل and که. The following few examples will hopefully shed some light on their usage.

In [18]:
# Creating a DF for مثل
مثل_bool = []
for line in sov_df['Far']:
    if 'مثل' in line:
        مثل_bool.append(True)
    else:
        مثل_bool.append(False)
has_مثل = pd.Series(مثل_bool)

In [19]:
مثل_sov_df = sov_df[مثل_bool]

Again, let's examine what the data looks like and then move check out some example sentences:

In [20]:
مثل_sov_df.tail(3)
len(مثل_sov_df)

ID                                                Eng  \
120366  110932  kids  you like a famiiy say no to where he is ...   
120422  128020  been leading me  like a thread  throughout the...   
120423  128019             had not its volatile  needlelike sword   

                                                      Far  \
120366  بچهها ، شما مثل يک خانواده ميمونيد نگو مثل خان...   
120422             مرا مثل قلابي در ميان جهان آويزان کرده   
120423                  لطا فتش را مثل شمشير از د ست داده   

                                                  Eng_Tok  \
120366  [kids, you, like, a, famiiy, say, no, to, wher...   
120422  [been, leading, me, like, a, thread, throughou...   
120423       [had, not, its, volatile, needlelike, sword]   

                                                  Far_Tok  Eng_Len  Far_Len  \
120366  [بچهها, ،, شما, مثل, يک, خانواده, ميمونيد, نگو...       12       14   
120422    [مرا, مثل, قلابي, در, ميان, جهان, آويزان, کرده]        9        8   
120423        [لطا, فتش, را, مثل, شمشير, از, د, ست, داده]        6        9   

                                                Eng_Types  \
120366  {to, he, say, is, famiiy, like, kids, no, wher...   
120422  {the, been, throughout, thread, world, like, a...   
120423       {its, volatile, sword, needlelike, had, not}   

                                                Far_Types  \
120366  {ميشه, ،, خانواده, يک, ديو, شما, ناراحت, مثل, ...   
120422    {آويزان, در, مثل, جهان, ميان, قلابي, کرده, مرا}   
120423        {د, از, داده, شمشير, لطا, مثل, ست, فتش, را}   

                                                  Far_POS  \
120366  [(بچهها, N), (،, PUNC), (شما, PRO), (مثل, ADVe...   
120422  [(مرا, PRO), (مثل, ADVe), (قلابي, N), (در, P),...   
120423  [(لطا, N), (فتش, N), (را, POSTP), (مثل, ADVe),...   

                                               Far_Chunks  \
120366  [بچهها NP] ، [شما NP] [مثل PP] [يک خانواده ميم...   
120422  [مرا NP] [مثل PP] [قلابي NP] [در PP] [ميان جها...   
120423  [لطا NP] [فتش NP] [را POSTP] [مثل PP] [شمشير N...   

                                                  Eng_POS  \
120366  [(kids, NNS), (you, PRP), (like, IN), (a, DT),...   
120422  [(been, VBN), (leading, VBG), (me, PRP), (like...   
120423  [(had, VBD), (not, RB), (its, PRP$), (volatile...   

                                               Eng_Chunks Word_Order_Final  
120366  [(kids, NNS), (you, PRP), [[('like', 'IN')], [...              SOV  
120422  [[[('been', 'VBN')]], [[('leading', 'VBG')]], ...              SOV  
120423  [[[('had', 'VBD')]], (not, RB), (its, PRP$), [...              SOV

1599

There's a _lot_ more entries for this one, compared to the "is like" DF. That being said, this may be because مثل is often used as a connector in common speech. 

In [21]:
for sample in مثل_sov_df['Far_Chunks'].sample(5):
    print(sample)
    print()

: [من NP] [مثل PP] [همه NP] [به PP] [چهار دليل NP] [دكتر ADJP] [شدم VP]

و [آيا ما NP] [را POSTP] [مثل PP] [هميشه تقديس NP] [خواهي VP] كرد

[کتابها NP] ، [مثل PP] [يک پناهگاه NP] [براي VP] [روياهاي PP] [ما NP] [هستند VP]

[لشگر سرخ NP] ! ، [به PP] [لشگر سفيد، NP] [با PP] [آرايش مثلثي NP] [حمله کنيد VP]

[اون ور NP] [مثل PP] [تو NP] [در PP] [قلب من NP] [هست VP]



In these instances of مثل, "like" is used less as a simile but more as an interjection, in a similar fashion to the stereotype of speech patterns for Californians -- the first sentence  translates to "I was like, 'I have four reasons to become a doctor." What does که look like? We'll follow the usual pattern for creating a DF off a Series for که.

In [22]:
# Creating a new DF around که
که_bool = []
for line in sov_df['Far']:
    if ' که ' in line:
        که_bool.append(True)
    else:
        که_bool.append(False)
has_که = pd.Series(مثل_bool)

In [23]:
که_sov_df = sov_df[که_bool]

In [24]:
که_sov_df.head()
len(که_sov_df)

ID                                                Eng  \
8   261660          it s you oh  my god it s everyone at here   
10  261656  i think we shouid get together and taik about ...   
14  261648                        it s aii i need to remember   
16  261640  dont we have a speciai bottie of something spa...   
21  261627   what i hope is that we can go on iiving together   

                                                  Far  \
8   اين تويي ، اوه خداي من اين هرکسي هست که اينجا هست   
10  من فکر ميکنم ما بايد با هم بمونيم وراجع به چيز...   
14               اين همه چيزيه که ميخوام به ياد بيارم   
16  يه بطري مخصوص از چيزي که براي مهمون هامون برق ...   
21  چيزي که من اميدوارم اينه که ما ميتونيم با هم ز...   

                                              Eng_Tok  \
8   [it, s, you, oh, my, god, it, s, everyone, at,...   
10  [i, think, we, shouid, get, together, and, tai...   
14                [it, s, aii, i, need, to, remember]   
16  [dont, we, have, a, speciai, bottie, of, somet...   
21  [what, i, hope, is, that, we, can, go, on, iiv...   

                                              Far_Tok  Eng_Len  Far_Len  \
8   [اين, تويي, ،, اوه, خداي, من, اين, هرکسي, هست,...       11       12   
10  [من, فکر, ميکنم, ما, بايد, با, هم, بمونيم, ورا...       14       17   
14      [اين, همه, چيزيه, که, ميخوام, به, ياد, بيارم]        7        8   
16  [يه, بطري, مخصوص, از, چيزي, که, براي, مهمون, ه...       12       12   
21  [چيزي, که, من, اميدوارم, اينه, که, ما, ميتونيم...       11       12   

                                            Eng_Types  \
8       {here, my, it, s, god, at, you, everyone, oh}   
10  {and, to, think, together, need, about, we, wh...   
14                {to, need, it, aii, remember, s, i}   
16  {bottie, of, for, dont, speciai, we, guests, s...   
21  {can, iiving, together, on, we, that, is, what...   

                                            Far_Types  \
8   {هست, خداي, ،, من, هرکسي, اين, که, اوه, تويي, ...   
10  {بايد, بديم, وراجع, ما, بمونيم, من, کنيم, ميکن...   
14      {ميخوام, ياد, بيارم, اين, که, چيزيه, همه, به}   
16  {از, هامون, مخصوص, مهمون, يه, نداريم, برق, که,...   
21  {زندگي, اينه, ما, من, کنيم, ميتونيم, با, هم, ک...   

                                              Far_POS  \
8   [(اين, Ne), (تويي, N), (،, PUNC), (اوه, RES), ...   
10  [(من, PRO), (فکر, Ne), (ميکنم, Ne), (ما, PRO),...   
14  [(اين, DET), (همه, DET), (چيزيه, N), (که, CONJ...   
16  [(يه, CONJ), (بطري, Ne), (مخصوص, AJ), (از, P),...   
21  [(چيزي, N), (که, CONJ), (من, PRO), (اميدوارم, ...   

                                           Far_Chunks  \
8   [اين تويي NP] ، [اوه خداي من NP] [اين هرکسي NP...   
10  [من NP] [فکر ميکنم ما NP] [بايد VP] [با PP] [ه...   
14  [اين همه چيزيه NP] که [ميخوام NP] [به PP] [ياد...   
16  [يه بطري مخصوص NP] [از PP] [چيزي NP] که [براي ...   
21  [چيزي NP] که [من NP] [اميدوارم VP] [اينه NP] ک...   

                                              Eng_POS  \
8   [(it, PRP), (s, VBZ), (you, PRP), (oh, VBP), (...   
10  [(i, JJ), (think, VBP), (we, PRP), (shouid, VB...   
14  [(it, PRP), (s, VBD), (aii, NN), (i, NNS), (ne...   
16  [(dont, NN), (we, PRP), (have, VBP), (a, DT), ...   
21  [(what, WP), (i, NN), (hope, VBP), (is, VBZ), ...   

                                           Eng_Chunks Word_Order_Final  
8   [(it, PRP), [[('s', 'VBZ')]], (you, PRP), [[('...              SOV  
10  [[(i, JJ)], [[('think', 'VBP')]], (we, PRP), [...              SOV  
14  [(it, PRP), [[('s', 'VBD')], [('aii', 'NN')]],...              SOV  
16  [[(dont, NN)], (we, PRP), [[('have', 'VBP')], ...              SOV  
21  [(what, WP), [(i, NN)], [[('hope', 'VBP')]], [...              SOV

16074

This one is _much, much_ bigger than either of the two previous DFs, but that's to be expected, since که is one of the most common conjunctive words. What do some of the sentences look like?

In [25]:
for sample in که_sov_df['Far_Chunks'].sample(5):
    print(sample)
    print()

[آيا آنها NP] [سر حرفشون NP] [ميايستند VP] وقتي که [راي آوردند VP]

[ميتوني NP] [به PP] [چيزي NP] که [دوست داري VP] ، [ادامه بدي NP]

[شوری NP] ، [وقتی ADVP] که [غذا NP] [سرد ADJP] [میشود VP] [به PP] [خوبی NP] [احساس میشود VP]

[پيش خودمون ميمونه NP] ، [اونجا ADVP] [يک NP] [کاميونه VP] که [توش NP] [دوربينه ADVP] [توي لندن NP] [تو NP] [جودي VP] دينچ

[وقتي NP] که [شاه مرد اون NP] [شاهه بعدي VP] ميشه



Unfortunately, که is not used abstractly whatsoever in these functions. In an ideal world, I would narrow the environments that که appears in to only those within similes and metaphors; due to the limitations of my coding abilities, I do not believe that I could have accomplished such a task within the time provided. 

Regardless, let's move on to the SVO data to see if there's anything meaningful that can be drawn from it.

### Section 3: Examining SVO Data
Similar to what we did for the SOV data, let's flash a little bit of what we're working with, then modify it in an identical way so as to keep things consistent:

In [26]:
svo_df.head()

ID                                                Eng  \
143410  416676       - i'd get going now. - ok see you next time.   
162921  476284  it is fortunate that i followed mother's words...   
169165  495157  leaving buyeo to establish a new nation means ...   
185577  544000  chunghae boasts the strongest navy in the sout...   
185704  544253  we must take yi do hyung to the imperial city ...   

                                                      Far  \
143410           - من ديگه بايد بريم- باشه، بعدا مي بينمت   
162921  خيلي خوب شد كه به حرفهاي مادر گوش كردم و به جن...   
169165  ترك بويه يو براي تاسيس يك طايفه ي جديد !معنيش ...   
185577  کشتي هاي کوچک چانگ هي قوي ترين نيروي دريايي جن...   
185704  ما باید همین حالا یی دو-یانگ را به شهر سلطنتی ...   

                                                  Eng_Tok  \
143410  [-, i, 'd, get, going, now, ., -, ok, see, you...   
162921  [it, is, fortunate, that, i, followed, mother,...   
169165  [leaving, buyeo, to, establish, a, new, nation...   
185577  [chunghae, boasts, the, strongest, navy, in, t...   
185704  [we, must, take, yi, do, hyung, to, the, imper...   

                                                  Far_Tok  Eng_Len  Far_Len  \
143410  [-, من, ديگه, بايد, بريم-, باشه،, بعدا, مي, بي...       14        9   
162921  [خيلي, خوب, شد, كه, به, حرفهاي, مادر, گوش, كرد...       14       13   
169165  [ترك, بويه, يو, براي, تاسيس, يك, طايفه, ي, جدي...       17       19   
185577  [کشتي, هاي, کوچک, چانگ, هي, قوي, ترين, نيروي, ...       10       16   
185704  [ما, باید, همین, حالا, یی, دو-یانگ, را, به, شه...       13       11   

                                                Eng_Types  \
143410  {you, ok, going, -, next, get, now, 'd, see, i...   
162921  {and, went, followed, that, battle, into, it, ...   
169165  {'re, enemy, you, going, buyeo, to, !, establi...   
185577  {the, boasts, navy, strongest, in, seas, chung...   
185704  {city, the, must, right, to, take, we, hyung, ...   

                                                Far_Types  \
143410  {بايد, -, من, بعدا, بريم-, ديگه, مي, باشه،, بي...   
162921  {جنگ, گوش, رفتم, مادر, كردم, خيلي, و, به, خوب,...   
169165  {يك, يو, !, معنيش, اينه, باشي, خواي, ي, تاسيس,...   
185577  {ترين, کشتي, رو, چانگ, هاي, غربي, هي, ميده, کو...   
185704  {را, حالا, یی, ما, باید, شهر, سلطنتی, دو-یانگ,...   

                                                  Far_POS  \
143410  [(-, PUNC), (من, PRO), (ديگه, N), (بايد, V), (...   
162921  [(خيلي, N), (خوب, AJ), (شد, V), (كه, CONJ), (ب...   
169165  [(ترك, N), (بويه, V), (يو, CONJ), (براي, P), (...   
185577  [(کشتي, Ne), (هاي, Ne), (کوچک, AJe), (چانگ, N)...   
185704  [(ما, PRO), (باید, V), (همین, DET), (حالا, ADV...   

                                               Far_Chunks  \
143410  - [من NP] [ديگه بايد VP] [بريم- PP] [باشه، بعد...   
162921  [خيلي NP] [خوب ADJP] [شد VP] كه [به PP] [حرفها...   
169165  [ترك NP] [بويه VP] يو [براي PP] [تاسيس يك طايف...   
185577  [کشتي هاي کوچک چانگ هي قوي ترين NP] [نيروي VP]...   
185704  [ما NP] [باید VP] [همین حالا ADVP] [یی NP] [دو...   

                                                  Eng_POS  \
143410  [(-, :), (i, NN), ('d, MD), (get, VB), (going,...   
162921  [(it, PRP), (is, VBZ), (fortunate, JJ), (that,...   
169165  [(leaving, VBG), (buyeo, NN), (to, TO), (estab...   
185577  [(chunghae, NN), (boasts, VBZ), (the, DT), (st...   
185704  [(we, PRP), (must, MD), (take, VB), (yi, NN), ...   

                                               Eng_Chunks Word_Order_Final  
143410  [(-, :), [(i, NN)], ('d, MD), [[('get', 'VB')]...              SVO  
162921  [(it, PRP), [[('is', 'VBZ')], [('fortunate', '...              SVO  
169165  [[[('leaving', 'VBG')], [('buyeo', 'NN')]], (t...              SVO  
185577  [[(chunghae, NN)], [[('boasts', 'VBZ')], [('th...              SVO  
185704  [(we, PRP), (must, MD), [[('take', 'VB')], [('...              SVO

In [27]:
# Making the indexes easier to work with
svo_df.reset_index(drop = True, inplace = True)

In [28]:
# Calculating average word length for the English translations of the SVO sentences
avg_eng_len2 = []
for ln in svo_df['Eng_Len']:
    avg_eng_len2.append(ln)
print('The average English sentence length in the SOV file is...', (sum(avg_eng_len2))/len(svo_df))

The average English sentence length in the SOV file is... 9.187980392725585


In [29]:
# Calculating average word length for the Persian translation of the SVO sentences
avg_far_len2 = []
for ln in svo_df['Far_Len']:
    avg_far_len2.append(ln)
print('The average Persian sentence length in the SOV file is...', (sum(avg_far_len2))/len(svo_df))

The average Persian sentence length in the SOV file is... 9.0338341503031


Like we searched through the English SOV sentences for "is like", let's do the same for the SVO sentences!

In [30]:
is_like_bool2 = []
for line in svo_df['Eng']:
    if 'is like' in line:
        is_like_bool2.append(True)
    else:
        is_like_bool2.append(False)
has_like2 = pd.Series(is_like_bool2)

In [56]:
len(svo_df)

68954

In [31]:
like_svo_df = svo_df[has_like2]

In [32]:
print(len(like_svo_df))
like_svo_df.head()

26


ID                                                Eng  \
3931   552576  the enemy you must fight is like a huge maelst...   
15324  497782  i heard that the coalition is like brothers of...   
16471  434104               rounds is like being on a game show.   
16518  434767                         that song is like a virus.   
19827  129997             your wife is like to reap a proper man   

                                                     Far  \
3931   اون دشمني که شما ميخواين باهاش بجنگين مثل يک گ...   
15324      من شنيدم که اين گروه مثل برادر هاي خوني هستند   
16471          اين ويزيتها شبيه اغاز يه مسابقه تلوزيونيه   
16518                         اين آهنگه مثله خوره ميمونه   
19827  همسر شما خوش‌اقبال است كه دوباره به شوهري دلخو...   

                                                 Eng_Tok  \
3931   [the, enemy, you, must, fight, is, like, a, hu...   
15324  [i, heard, that, the, coalition, is, like, bro...   
16471    [rounds, is, like, being, on, a, game, show, .]   
16518                [that, song, is, like, a, virus, .]   
19827   [your, wife, is, like, to, reap, a, proper, man]   

                                                 Far_Tok  Eng_Len  Far_Len  \
3931   [اون, دشمني, که, شما, ميخواين, باهاش, بجنگين, ...       11       11   
15324  [من, شنيدم, که, اين, گروه, مثل, برادر, هاي, خو...       12       10   
16471  [اين, ويزيتها, شبيه, اغاز, يه, مسابقه, تلوزيونيه]        9        7   
16518                   [اين, آهنگه, مثله, خوره, ميمونه]        7        5   
19827  [همسر, شما, خوش‌اقبال, است, كه, دوباره, به, شو...        9       11   

                                               Eng_Types  \
3931   {the, must, maelstrom, enemy, fight, is, huge,...   
15324  {the, blood, of, that, is, like, one, brothers...   
16471    {rounds, game, being, on, is, like, show, a, .}   
16518                {virus, that, is, like, song, a, .}   
19827   {to, proper, your, reap, is, like, wife, a, man}   

                                               Far_Types  \
3931   {دشمني, اون, يک, شما, بزرگه, ميخواين, که, مثل,...   
15324  {خوني, من, شنيدم, هاي, گروه, هستند, برادر, اين...   
16471  {مسابقه, شبيه, يه, ويزيتها, اين, اغاز, تلوزيونيه}   
16518                   {ميمونه, خوره, اين, مثله, آهنگه}   
19827  {است, همسر, خوش‌اقبال, مي‌يابد, دوباره, شما, د...   

                                                 Far_POS  \
3931   [(اون, DET), (دشمني, V), (که, CONJ), (شما, PRO...   
15324  [(من, PRO), (شنيدم, V), (که, CONJ), (اين, Ne),...   
16471  [(اين, DET), (ويزيتها, N), (شبيه, V), (اغاز, N...   
16518  [(اين, N), (آهنگه, V), (مثله, Ne), (خوره, Ne),...   
19827  [(همسر, Ne), (شما, PRO), (خوش‌اقبال, AJ), (است...   

                                              Far_Chunks  \
3931   [اون NP] [دشمني VP] که [شما NP] [ميخواين ADJP]...   
15324  [من NP] [شنيدم VP] که [اين گروه NP] [مثل PP] [...   
16471  [اين ويزيتها NP] [شبيه VP] [اغاز يه مسابقه تلو...   
16518          [اين NP] [آهنگه VP] [مثله خوره ميمونه NP]   
19827  [همسر شما NP] [خوش‌اقبال ADJP] [است VP] كه [دو...   

                                                 Eng_POS  \
3931   [(the, DT), (enemy, NN), (you, PRP), (must, MD...   
15324  [(i, RB), (heard, VBP), (that, IN), (the, DT),...   
16471  [(rounds, NNS), (is, VBZ), (like, IN), (being,...   
16518  [(that, DT), (song, NN), (is, VBZ), (like, IN)...   
19827  [(your, PRP$), (wife, NN), (is, VBZ), (like, I...   

                                              Eng_Chunks Word_Order_Final  
3931   [[(the, DT), (enemy, NN)], (you, PRP), (must, ...              SVO  
15324  [(i, RB), [[('heard', 'VBP')], [(P that/IN), (...              SVO  
16471  [(rounds, NNS), [[('is', 'VBZ')]], [(like, IN)...              SVO  
16518  [[(that, DT), (song, NN)], [[('is', 'VBZ')], [...              SVO  
19827  [(your, PRP$), [(wife, NN)], [[('is', 'VBZ')]]...              SVO

Proportionally, there are less "is like" sentences in the SVO data than there are in the SOV data:
- 56/120607 = 0.047% for the SOV data
- 28/68954 = 0.038% for the SVO data

As mentioned before, مثل and که are the functional words here that are often used to replace "is like" in English. However, only three of the sentences in this DF use those functional words! These sentences still translate the idea of a simile, but convert it to a more directly metaphorical approach. This approach may have an influence on why the structure of the translation is SVO...! However, this is definitely not a conclusion I can make quite yet -- there's much, _much_ more to do! For now, let's look at how the مثل and که are used! 

In [33]:
# Creating the مثل analysis
مثل_bool2 = []
for line in svo_df['Far']:
    if 'مثل' in line:
        مثل_bool2.append(True)
    else:
        مثل_bool2.append(False)
has_مثل2 = pd.Series(مثل_bool2)

In [34]:
مثل_svo_df = svo_df[مثل_bool2]

In [35]:
مثل_svo_df.tail(3)

ID                                                Eng  \
68888  270971  you above all people shouldnt be one to dismis...   
68898  270891  few realize that mary was descended from kings...   
68899  270889  peter  i see you contending against a woman li...   

                                                     Far  \
68888  آدم هاي مثل تو نبايد کسي باشندکه نفوذ گذشته را...   
68898  عده کمي ميدونند که مريمثل شوهرش از نسل پادشاها...   
68899   پيتر ، ميبينم که تو عليهيک زن مثل يک حريف ميجنگي   

                                                 Eng_Tok  \
68888  [you, above, all, people, shouldnt, be, one, t...   
68898  [few, realize, that, mary, was, descended, fro...   
68899  [peter, i, see, you, contending, against, a, w...   

                                                 Far_Tok  Eng_Len  Far_Len  \
68888  [آدم, هاي, مثل, تو, نبايد, کسي, باشندکه, نفوذ,...       14       13   
68898  [عده, کمي, ميدونند, که, مريمثل, شوهرش, از, نسل...       13       10   
68899  [پيتر, ،, ميبينم, که, تو, عليهيک, زن, مثل, يک,...       11       11   

                                               Eng_Types  \
68888  {dismiss, the, to, influence, of, shouldnt, pa...   
68898  {her, mary, kings, as, husband, descended, tha...   
68899  {contending, you, adversary, against, peter, l...   

                                               Far_Types  \
68888  {از, کسي, نفوذ, هاي, تو, دست, باشندکه, بدهند, ...   
68898  {پادشاهان, از, کمي, نسل, شوهرش, بوده, که, مريم...   
68899  {،, عليهيک, ميبينم, زن, يک, تو, پيتر, حريف, که...   

                                                 Far_POS  \
68888  [(آدم, N), (هاي, V), (مثل, ADVe), (تو, PRO), (...   
68898  [(عده, N), (کمي, AJ), (ميدونند, V), (که, CONJ)...   
68899  [(پيتر, N), (،, PUNC), (ميبينم, V), (که, CONJ)...   

                                              Far_Chunks  \
68888  [آدم NP] [هاي VP] [مثل PP] [تو NP] [نبايد VP] ...   
68898  [عده NP] [کمي ADJP] [ميدونند VP] که [مريمثل شو...   
68899  [پيتر NP] ، [ميبينم VP] که [تو NP] [عليهيک زن ...   

                                                 Eng_POS  \
68888  [(you, PRP), (above, IN), (all, DT), (people, ...   
68898  [(few, JJ), (realize, NN), (that, IN), (mary, ...   
68899  [(peter, NN), (i, NN), (see, VBP), (you, PRP),...   

                                              Eng_Chunks Word_Order_Final  
68888  [(you, PRP), [[('above', 'IN')], [('all', 'DT'...              SVO  
68898  [[(few, JJ), (realize, NN)], [[('that', 'IN')]...              SVO  
68899  [[(peter, NN), (i, NN)], [[('see', 'VBP')]], (...              SVO

1599

What do some sentences look like?

In [36]:
for sample in مثل_svo_df['Far_Chunks'].sample(5):
    print(sample)
    print()

[هرمويي NP] که [بيرون بمونه VP] [مثل PP] [خنجريه NP] که [تو NP] [قلب NP] [شهيد VP] [ما NP] [فرو ميره ADVP]

[اون مرده NP] [خوبيه VP] يک [چيزي مثله تو NP]

[سر من NP] [داره VP] [مثل PP] [گرد NP] ، [تو NP] [طوفان شن ميچرخه NP] [بايد VP] [برم VP]

[شما NP] [نميتونيد VP] [تک تک مردم NP] [رومثل PP] [ما NP] [متقاعد ADJP] [کنيد VP]

[اونا NP] [نمي VP] [توننمثل ايول مي ايول پيش NP] [بيني PP] [کنن NP] .



The odd thing about this data is that it does not always accurately predict the phrasal structure of مثل. Sometimes, as in the first, third, and fourth sentences, مثل is treated as a PP (though the fourth sentences has a slightly modified version of the word). The second and fifth sentences mark مثل as part of a NP. This irregularity between chunks is an example of a limitation of the chunker. 

The sentences themselves do not contain any metaphorical uses of مثل, although it clear that they do exist in some fashion (as proven by the previous search for "is like"). Let's see if any interesting points can be drawn from looking at که. 

In [37]:
# Looking at که, again!
که_bool2 = []
for line in svo_df['Far']:
    if ' که ' in line:
        که_bool2.append(True)
    else:
        که_bool2.append(False)
has_که = pd.Series(مثل_bool)

In [38]:
که_svo_df = svo_df[که_bool2]

In [39]:
که_svo_df.head()
len(که_svo_df)

ID                                                Eng  \
40  553945  i know reform is difficult as long as there ar...   
50  554238              you said you wished me to be happy...   
67  485809        i put everything l had in this transaction!   
83  554000                     i believe you will protect her   
84  553929              i was forced to commit a grave wrong.   

                                                  Far  \
40  من ميدونم که تا زماني که صاحب منصبان هستند اصل...   
50                     تو گفتي که ميخواهي من شاد باشم   
67                هر چي که داشتم تو اين معامله گذاشتم   
83                من ميدونم که تو از اون مراقبت ميکني   
84        من مجبور شدم که اين کار اشتباه رو انجام بدم   

                                              Eng_Tok  \
40  [i, know, reform, is, difficult, as, long, as,...   
50   [you, said, you, wished, me, to, be, happy, ...]   
67  [i, put, everything, l, had, in, this, transac...   
83              [i, believe, you, will, protect, her]   
84   [i, was, forced, to, commit, a, grave, wrong, .]   

                                              Far_Tok  Eng_Len  Far_Len  \
40  [من, ميدونم, که, تا, زماني, که, صاحب, منصبان, ...       11       13   
50             [تو, گفتي, که, ميخواهي, من, شاد, باشم]        9        7   
67       [هر, چي, که, داشتم, تو, اين, معامله, گذاشتم]        9        8   
83       [من, ميدونم, که, تو, از, اون, مراقبت, ميکني]        6        8   
84  [من, مجبور, شدم, که, اين, کار, اشتباه, رو, انج...        9       10   

                                            Eng_Types  \
40  {long, as, difficult, are, is, nobles, know, r...   
50        {to, said, wished, happy, be, you, ..., me}   
67  {transaction, put, this, !, l, everything, in,...   
83              {her, will, believe, protect, you, i}   
84   {wrong, to, was, commit, grave, a, i, ., forced}   

                                            Far_Types  \
40  {زماني, مشکل, است, کردن, تا, من, ميدونم, هستند...   
50             {باشم, من, تو, گفتي, شاد, که, ميخواهي}   
67       {تو, اين, که, گذاشتم, چي, هر, داشتم, معامله}   
83       {اون, از, من, تو, ميدونم, که, مراقبت, ميکني}   
84  {مجبور, من, رو, کار, شدم, انجام, اين, که, اشتب...   

                                              Far_POS  \
40  [(من, PRO), (ميدونم, V), (که, CONJ), (تا, P), ...   
50  [(تو, PRO), (گفتي, V), (که, CONJ), (ميخواهي, N...   
67  [(هر, DET), (چي, N), (که, CONJ), (داشتم, V), (...   
83  [(من, PRO), (ميدونم, V), (که, CONJ), (تو, PRO)...   
84  [(من, PRO), (مجبور, AJ), (شدم, V), (که, CONJ),...   

                                           Far_Chunks  \
40  [من NP] [ميدونم VP] که [تا PP] [زماني NP] که [...   
50  [تو NP] [گفتي VP] که [ميخواهي من NP] [شاد ADJP...   
67  [هر چي NP] که [داشتم VP] [تو NP] [اين معامله N...   
83  [من NP] [ميدونم VP] که [تو NP] [از PP] [اون NP...   
84  [من NP] [مجبور ADJP] [شدم VP] که [اين کار NP] ...   

                                              Eng_POS  \
40  [(i, NN), (know, VBP), (reform, NN), (is, VBZ)...   
50  [(you, PRP), (said, VBD), (you, PRP), (wished,...   
67  [(i, NN), (put, VBD), (everything, NN), (l, NN...   
83  [(i, NN), (believe, VBP), (you, PRP), (will, M...   
84  [(i, NN), (was, VBD), (forced, VBN), (to, TO),...   

                                           Eng_Chunks Word_Order_Final  
40  [[(i, NN)], [[('know', 'VBP')], [('reform', 'N...              SVO  
50  [(you, PRP), [[('said', 'VBD')]], (you, PRP), ...              SVO  
67  [[(i, NN)], [[('put', 'VBD')], [('everything',...              SVO  
83  [[(i, NN)], [[('believe', 'VBP')]], (you, PRP)...              SVO  
84  [[(i, NN)], [[('was', 'VBD')]], [[('forced', '...              SVO

11227

Proportionally, که is used much more in SVO sentences. I expected that this may occur, since the chunker sometimes does not include که in chunks. Because که is not always included but usually appears right after the verb, a simple SV or OV sentence may be seen as a complex sentence. This is a result of not searching for a specific context of که when creating this DF.

What do some of the SVO sentences using که look like?

In [40]:
for sample in که_svo_df['Far_Chunks'].sample(5):
    print(sample)
    print()

[گارد سلطنتي NP] [تقاضا دارن VP] که [دوشيزه ADVP] [سو NP] [سئو-نو NP] [رو POSTP] [ملاقات کنند VP] .

[تو NP] که [ديدي VP] [من تو NP] [واگن اولي NP] [هستم VP] [چرا ADVP] [رفتي VP] [تو NP] [دومي‌ نشستي VP]

[من NP] [میدونم VP] که [شما NP] [در PP] [مورد این NP] ، [احساس خوبی NP] [ندارید VP] [به PP] [دلیل اتفاقی NP] که [دفعه NP] [پیش افتاد VP]

[پنجاه هزار تا NP] [دارم VP] که [ميگه ميتوني کمکم NP] [کني VP]

[نميتونم ژنرالي NP] [را POSTP] [پيدا کنم VP] که [جرات مواجه NP] [با PP] [هيتلر NP] [را POSTP] [داشته VP] [باشه VP]



Again, these sentences don't provide much support for که being a predictor of any particular word ordering. An interesting pattern can be found in the second sentence, which roughly translates to "You that I saw for the first time...", using an embedded syntactic structure that follows something like "NP that VP... VP". Since که is not tagged, the regular expressions limited this to the initial "NP VP", but the real structure would just be "NP".

These two word-orders didn't tell us very much, unfortunately -- what might the "Other" data say (pardon my pun)?

### Section 4: Examining Other Data
Let's load in the Other data and see what it has to offer:

In [41]:
other_df = pd.read_pickle('only_other_df.pkl')

In [42]:
other_df.head()
len(other_df)

ID                 Eng                 Far                  Eng_Tok  \
0   1     raspy breathing          صداي خر خر       [raspy, breathing]   
1   2                 dad                 پدر                    [dad]   
2   3  maybe its the wind  شايد صداي باد باشه  [maybe, its, the, wind]   
3   4                  no                  نه                     [no]   
6   7             william              ويليام                [william]   

                   Far_Tok  Eng_Len  Far_Len                Eng_Types  \
0           [صداي, خر, خر]        2        3       {breathing, raspy}   
1                    [پدر]        1        1                    {dad}   
2  [شايد, صداي, باد, باشه]        4        4  {its, the, wind, maybe}   
3                     [نه]        1        1                     {no}   
6                 [ويليام]        1        1                {william}   

                 Far_Types                                        Far_POS  \
0               {خر, صداي}               [(صداي, NUM), (خر, Ne), (خر, N)]   
1                    {پدر}                                     [(پدر, N)]   
2  {باشه, صداي, باد, شايد}  [(شايد, Ne), (صداي, AJ), (باد, V), (باشه, V)]   
3                     {نه}                                    [(نه, ADV)]   
6                 {ويليام}                                  [(ويليام, N)]   

                          Far_Chunks  \
0                    [صداي خر خر NP]   
1                           [پدر NP]   
2  [شايد صداي NP] [باد VP] [باشه VP]   
3                                 نه   
6                        [ويليام NP]   

                                             Eng_POS  \
0                     [(raspy, NN), (breathing, NN)]   
1                                        [(dad, NN)]   
2  [(maybe, RB), (its, PRP$), (the, DT), (wind, NN)]   
3                                         [(no, DT)]   
6                                    [(william, NN)]   

                                          Eng_Chunks Word_Order_Final  
0                   [[(raspy, NN), (breathing, NN)]]            Other  
1                                      [[(dad, NN)]]            Other  
2  [(maybe, RB), (its, PRP$), [(the, DT), (wind, ...            Other  
3                                       [[(no, DT)]]            Other  
6                                  [[(william, NN)]]            Other

422525

In [43]:
# Making the indexes easier to work with
other_df.reset_index(drop = True, inplace = True)

In [44]:
# Checking average sentence length
avg_eng_len3 = []
for ln in other_df['Eng_Len']:
    avg_eng_len3.append(ln)
print('The average English sentence length in the other file is...', (sum(avg_eng_len3))/len(other_df))

The average English sentence length in the other file is... 5.751316490148512


In [45]:
# Checking average sentence length
avg_far_len3 = []
for ln in other_df['Far_Len']:
    avg_far_len3.append(ln)
print('The average Persian sentence length in the other file is...', (sum(avg_far_len3))/len(other_df))

The average Persian sentence length in the other file is... 5.347882373824034


Whoa! The average sentence length for both English and Persian is _much_ shorter! Of course, this is probably a result of the influx of one-word/two-word lines that proliferate this file.

Again, let's check out the "is like" structure.

In [46]:
is_like_bool3 = []
for line in other_df['Eng']:
    if 'is like' in line:
        is_like_bool3.append(True)
    else:
        is_like_bool3.append(False)
has_like3 = pd.Series(is_like_bool3)

In [47]:
like_other_df = other_df[has_like3]

In [48]:
print(len(like_other_df))
like_other_df.head()

103


ID                                                Eng  \
3979    5923                   it is like a completely new life   
11532  17526                   its not no  dear a tune is like:   
30363  46011  that is like saying  because someone gave thei...   
40178  60759  like this like his face all red his eyes buggi...   
40249  60871  arlene  you ve forgotten what hanging out with...   

                                                     Far  \
3979                             اين مثل يک زندگيه جديده   
11532                   نداره :نه عزيزم يک اهنگ هست مثله   
30363  به خاطر اين كه يك نفر به چيزي فكر كرده كه اون ...   
40178  مثل اينتمام صورتش قرمز شد چشماش از حدقه ، داشت...   
40249  آرلين ؛ يادت رفته گردش رفتن با جوليا جنگلي چطوريه   

                                                 Eng_Tok  \
3979            [it, is, like, a, completely, new, life]   
11532         [its, not, no, dear, a, tune, is, like, :]   
30363  [that, is, like, saying, because, someone, gav...   
40178  [like, this, like, his, face, all, red, his, e...   
40249  [arlene, you, ve, forgotten, what, hanging, ou...   

                                                 Far_Tok  Eng_Len  Far_Len  \
3979                       [اين, مثل, يک, زندگيه, جديده]        7        5   
11532         [نداره, :, نه, عزيزم, يک, اهنگ, هست, مثله]        9        8   
30363  [به, خاطر, اين, كه, يك, نفر, به, چيزي, فكر, كر...        9       14   
40178  [مثل, اينتمام, صورتش, قرمز, شد, چشماش, از, حدق...       11       12   
40249  [آرلين, ؛, يادت, رفته, گردش, رفتن, با, جوليا, ...       12       10   

                                               Eng_Types  \
3979            {it, life, like, is, new, a, completely}   
11532         {its, dear, is, :, like, no, a, tune, not}   
30363  {their, someone, saying, that, is, like, atten...   
40178  {buggin, his, this, red, like, face, out, eyes...   
40249  {arlene, is, julia, what, ve, like, with, hang...   

                                               Far_Types  \
3979                       {يک, زندگيه, اين, مثل, جديده}   
11532         {هست, نداره, يک, عزيزم, اهنگ, نه, مثله, :}   
30363  {اون, فكر, يك, كرده, خاطر, اين, كه, نفر, نميخو...   
40178  {قرمز, از, حدقه, ،, اينتمام, بيرون, داشت, ميزد...   
40249  {؛, جوليا, گردش, آرلين, يادت, چطوريه, با, رفتن...   

                                                 Far_POS  \
3979   [(اين, N), (مثل, ADVe), (يک, NUM), (زندگيه, Ne...   
11532  [(نداره, V), (:, PUNC), (نه, ADV), (عزيزم, N),...   
30363  [(به, P), (خاطر, Ne), (اين, N), (كه, V), (يك, ...   
40178  [(مثل, ADVe), (اينتمام, Ne), (صورتش, N), (قرمز...   
40249  [(آرلين, N), (؛, PUNC), (يادت, N), (رفته, V), ...   

                                              Far_Chunks  \
3979              [اين NP] [مثل PP] [يک زندگيه جديده NP]   
11532  [نداره VP] : [نه ADVP] [عزيزم NP] [يک اهنگ NP]...   
30363  [به PP] [خاطر اين NP] [كه VP] [يك نفر NP] [به ...   
40178  [مثل PP] [اينتمام صورتش NP] [قرمز ADJP] [شد VP...   
40249  [آرلين NP] ؛ [يادت رفته گردش رفتن NP] [با PP] ...   

                                                 Eng_POS  \
3979   [(it, PRP), (is, VBZ), (like, IN), (a, DT), (c...   
11532  [(its, PRP$), (not, RB), (no, DT), (dear, NN),...   
30363  [(that, DT), (is, VBZ), (like, IN), (saying, V...   
40178  [(like, IN), (this, DT), (like, IN), (his, PRP...   
40249  [(arlene, NN), (you, PRP), (ve, VBP), (forgott...   

                                              Eng_Chunks Word_Order_Final  
3979   [(it, PRP), [[('is', 'VBZ')], [(P like/IN), (N...            Other  
11532  [(its, PRP$), (not, RB), [(no, DT), (dear, NN)...            Other  
30363  [[(that, DT)], [[('is', 'VBZ')]], [(like, IN)]...            Other  
40178  [[[('like', 'IN')], [('this', 'DT')]], [(like,...            Other  
40249  [[(arlene, NN)], (you, PRP), [[('ve', 'VBP')]]...            Other

Not a ton of choices here, with only 103 entries. Examining the five sentences in the head, it appears that "is like" behaves in a more explanatory comparison sense, rather than a metaphorical or similic one. What about مثل?

In [49]:
مثل_bool3 = []
for line in other_df['Far']:
    if 'مثل' in line:
        مثل_bool3.append(True)
    else:
        مثل_bool3.append(False)
has_مثل3 = pd.Series(مثل_bool3)

In [50]:
مثل_other_df = other_df[مثل_bool3]

In [51]:
مثل_other_df.tail(3)

ID        Eng       Far      Eng_Tok      Far_Tok  Eng_Len  \
421777  611338     wifely  مثل زوجه     [wifely]  [مثل, زوجه]        1   
421952  611513  wiredrawn   مثل سيم  [wiredrawn]   [مثل, سيم]        1   
422043  611605  womanlike    مثل زن  [womanlike]    [مثل, زن]        1   

        Far_Len    Eng_Types    Far_Types                   Far_POS  \
421777        2     {wifely}  {زوجه, مثل}  [(مثل, ADVe), (زوجه, N)]   
421952        2  {wiredrawn}   {سيم, مثل}   [(مثل, ADVe), (سيم, N)]   
422043        2  {womanlike}    {زن, مثل}    [(مثل, ADVe), (زن, N)]   

                Far_Chunks            Eng_POS           Eng_Chunks  \
421777  [مثل PP] [زوجه NP]     [(wifely, RB)]       [(wifely, RB)]   
421952   [مثل PP] [سيم NP]  [(wiredrawn, NN)]  [[(wiredrawn, NN)]]   
422043    [مثل PP] [زن NP]  [(womanlike, NN)]  [[(womanlike, NN)]]   

       Word_Order_Final  
421777            Other  
421952            Other  
422043            Other

1599

Funnily enough, it almost looks like the tail is alphebatized, but it's not! Things work out weirdly sometimes. That being said, this tail demonstrates a common feature of Persian morphology: compound nouns incorporating function words. Therefore, the word for "womanlike" directly translates to "like woman", the word for "wifely" translates to "like wife", and the word for "wiredrawn" translates to "like wire". 

What do که sentences look like?

In [52]:
که_bool3 = []
for line in other_df['Far']:
    if ' که ' in line:
        که_bool3.append(True)
    else:
        که_bool3.append(False)
has_که = pd.Series(مثل_bool3)

In [53]:
که_other_df = other_df[که_bool3]

In [54]:
که_other_df.head()
len(که_other_df)

ID                                                Eng  \
26  49  well  i heard about a balled up whore named ch...   
31  55  well  will you look at all this you all spared...   
32  56  ive got to say  thought its probably cheaper j...   
42  70  well  tommy it seems that there was a pinkerto...   
43  71  now  i know what charlie told you  because we ...   

                                                  Far  \
26  خوب من درباره يک جنده گندي که اسمش چارلي پرينس...   
31  خوب يک نگاه به اينا ميکني همه چيزهايي که داري ...   
32  من بايد بگم که احتمالا بي ارزشتر از اونيه که م...   
42  خوب تامي به نظر ميرسه که داخل اون دليجان يک پي...   
43  الان ميدونم که چارلي بهت چي گفته براي اينکه ما...   

                                              Eng_Tok  \
26  [well, i, heard, about, a, balled, up, whore, ...   
31  [well, will, you, look, at, all, this, you, al...   
32  [ive, got, to, say, thought, its, probably, ch...   
42  [well, tommy, it, seems, that, there, was, a, ...   
43  [now, i, know, what, charlie, told, you, becau...   

                                              Far_Tok  Eng_Len  Far_Len  \
26  [خوب, من, درباره, يک, جنده, گندي, که, اسمش, چا...       11       11   
31  [خوب, يک, نگاه, به, اينا, ميکني, همه, چيزهايي,...       15       13   
32  [من, بايد, بگم, که, احتمالا, بي, ارزشتر, از, ا...       16       15   
42  [خوب, تامي, به, نظر, ميرسه, که, داخل, اون, دلي...       16       16   
43  [الان, ميدونم, که, چارلي, بهت, چي, گفته, براي,...       18       18   

                                            Eng_Types  \
26  {i, up, balled, well, about, named, princess, ...   
31  {will, well, this, expense, no, byron, at, you...   
32  {the, damn, cheaper, its, to, probably, say, r...   
42  {coach, well, quite, that, was, it, inside, se...   
43  {but, charlie, this, dont, we, rules, what, to...   

                                            Far_Types  \
26  {جنده, من, شنيدم, درباره, يک, خوب, گندي, که, پ...   
31  {نداره, نگاه, داري, يک, هيچ, خوب, مصرفي, چيزها...   
32  {بايد, از, بدزدم, من, بي, اونيه, که, ارزشتر, ا...   
42  {ميرسه, اون, داخل, يود, دليجان, پينکرتون, تامي...   
43  {بهت, چندتا, قانون, ما, نداريم, تو, گروه, چارل...   

                                              Far_POS  \
26  [(خوب, ADV), (من, PRO), (درباره, Pe), (يک, Ne)...   
31  [(خوب, ADV), (يک, NUM), (نگاه, N), (به, P), (ا...   
32  [(من, PRO), (بايد, V), (بگم, V), (که, CONJ), (...   
42  [(خوب, ADV), (تامي, N), (به, P), (نظر, Ne), (م...   
43  [(الان, ADV), (ميدونم, V), (که, CONJ), (چارلي,...   

                                           Far_Chunks  \
26  [خوب ADJP] [من NP] [درباره PP] [يک جنده NP] [گ...   
31  [خوب ADJP] [يک نگاه NP] [به PP] [اينا NP] [ميک...   
32  [من NP] [بايد VP] [بگم VP] که [احتمالا ADVP] [...   
42  [خوب ADJP] [تامي NP] [به PP] [نظر ميرسه NP] که...   
43  [الان ADVP] [ميدونم VP] که [چارلي NP] [بهت PP]...   

                                              Eng_POS  \
26  [(well, RB), (i, RB), (heard, VBP), (about, IN...   
31  [(well, RB), (will, MD), (you, PRP), (look, VB...   
32  [(ive, JJ), (got, VBD), (to, TO), (say, VB), (...   
42  [(well, RB), (tommy, IN), (it, PRP), (seems, V...   
43  [(now, RB), (i, VBZ), (know, VBP), (what, WP),...   

                                           Eng_Chunks Word_Order_Final  
26  [(well, RB), (i, RB), [[('heard', 'VBP')], [(P...            Other  
31  [(well, RB), (will, MD), (you, PRP), [[('look'...            Other  
32  [[(ive, JJ)], [[('got', 'VBD')]], (to, TO), [[...            Other  
42  [(well, RB), [(tommy, IN)], (it, PRP), [[('see...            Other  
43  [(now, RB), [[('i', 'VBZ')]], [[('know', 'VBP'...            Other

24652

Again, the head here does not show much promise. Since there are nearly 25000 entries in this DF, it would be extremely difficult to conclusively narrow the relationship between که and "Other" word order.

### Section 5: Conclusions

Unfortunately, my data was not set-up well enough to provide a good analysis of grammatical functions. This is a result of a few issues, most notably an unreliable chunker and a lack of understanding of regular expressions.

Some general conclusions are hinted at, by are by no means confirmed:
- Phrases that contain the word "is like" do carry over their simile components, especially in SOV sentences
- SVO sentences do not appear to use مثل ("like") for metaphors in its usual equivalent position
- "که" needs to  

On an unrelated note, I will be here this summer, and would like to review what I could've done to improve my project. I think I bit off more than I could chew, and I'd like to take a second-look after the semester has concluded.